bundles/org.openhab.binding.homeconnect/src/main/java/org/openhab/binding/homeconnect/internal/servlet/HomeConnectServlet.java

```diff
@@ -0,0 +1,621 @@
+    private static final String ACTION_DOOR_STATE = "door-state";
+    private static final String ACTION_REMOTE_START_ALLOWED = "remote-control-start-allowed";
+    private static final String ACTION_REMOTE_CONTROL_ACTIVE = "remote-control-active";
+    private static final String ACTION_PUT_RAW = "put-raw";
+    private static final String ACTION_GET_RAW = "get-raw";
+    private static final DateTimeFormatter FILE_EXPORT_DTF = DateTimeFormatter.ofPattern("yyyy-MM-dd_HH-mm");
+    private static final long serialVersionUID = -2449763690208703307L;
+
+    private final Logger logger = LoggerFactory.getLogger(HomeConnectServlet.class);
+    private final HttpService httpService;
+    private final TemplateEngine templateEngine;
+    private final Set<HomeConnectBridgeHandler> bridgeHandlers;
+    private final Gson gson;
+
+    @Activate
+    public HomeConnectServlet(@Reference HttpService httpService) {
+        bridgeHandlers = new CopyOnWriteArraySet<>();
+        gson = new GsonBuilder().registerTypeAdapter(ZonedDateTime.class, (JsonSerializer<ZonedDateTime>) (src,
+                typeOfSrc, context) -> new JsonPrimitive(src.format(DateTimeFormatter.ISO_DATE_TIME))).create();
+        this.httpService = httpService;
+
+        // register servlet
+        try {
+            logger.info("Initialize Home Connect configuration servlet ({})", SERVLET_PATH);
+            httpService.registerServlet(SERVLET_PATH, this, null, httpService.createDefaultHttpContext());
+            httpService.registerResources(ASSETS_PATH, "assets", null);
+        } catch (NamespaceException e) {
+            try {
+                httpService.unregister(SERVLET_PATH);
+                httpService.unregister(ASSETS_PATH);
+                httpService.registerServlet(SERVLET_PATH, this, null, httpService.createDefaultHttpContext());
+                httpService.registerResources(ASSETS_PATH, "assets", null);
+            } catch (ServletException | NamespaceException ex) {
+                logger.error("Could not register Home Connect servlet! ({})", SERVLET_PATH, ex);
+            }
+        } catch (ServletException e) {
+            logger.error("Could not register Home Connect servlet! ({})", SERVLET_PATH, e);
+        }
+
+        // setup template engine
+        ServletContextTemplateResolver templateResolver = new ServletContextTemplateResolver(getServletContext());
+        templateResolver.setTemplateMode(TemplateMode.HTML);
+        templateResolver.setPrefix("/templates/");
+        templateResolver.setSuffix(".html");
+        templateResolver.setCacheable(true);
+        templateEngine = new TemplateEngine();
+        templateEngine.addDialect(new Java8TimeDialect());
+        templateEngine.setTemplateResolver(templateResolver);
+    }
+
+    @Deactivate
+    protected void dispose() {
+        try {
+            logger.info("Unregister Home Connect configuration servlet ({}).", SERVLET_PATH);
```

Comment:
Can this message be replaced by using the debugger or by increasing the framework's log level? See point 4 https://www.openhab.org/docs/developer/guidelines.html#f-logging